lightning-persister/src/util.rs

```diff
@@ -39,21 +32,22 @@ fn path_to_windows_str<T: AsRef<OsStr>>(path: T) -> Vec<winapi::shared::ntdef::W
 }
 
 #[allow(bare_trait_objects)]
-pub(crate) fn write_to_file<D: DiskWriteable>(path: PathBuf, filename: String, data: &D) -> std::io::Result<()> {
-	fs::create_dir_all(path.clone())?;
+pub(crate) fn write_to_file<W: Writeable>(filename_with_path: String, data: &W) -> std::io::Result<()> {
+	let full_path = PathBuf::from(&filename_with_path);
```

Comment:
If you move the `tmp_filename` creation to above here we can cut an allocation by dropping the reference here.

Reply:
Drop `&` as per above?